multicluster/controllers/multicluster/clusterset_controller.go

```diff
@@ -287,8 +268,23 @@ func (r *ClusterSetReconciler) updateMultiClusterSetOnMemberCluster(clusterSet *
 		klog.InfoS("Deleted RemoteCluster", "clusterID", remoteCluster.GetClusterID())
 	}
 
-	if err != nil {
-		klog.ErrorS(err, "Final error")
+	return multiErr
+}
+
+// getSecretForLeader returns the Secret associated with this local cluster(which is a member)
+// for the given leader.
+// When a member is added to a ClusterSet, a specific ServiceAccount is created on the
+// leader cluster which allows the member access into the CommonArea. This ServiceAccount
+// has an associated Secret which must be copied into the member cluster as an opaque secret.
+// Name of this secret is part of the ClusterSet spec for this leader. This method reads
+// the Secret given by that name.
+func (r *ClusterSetReconciler) getSecretForLeader(secretName string, secretNs string) (secretObj *v1.Secret) {
+	secretNamespacedName := types.NamespacedName{
+		Namespace: secretNs,
+		Name:      secretName,
+	}
+	if err := r.Get(context.TODO(), secretNamespacedName, secretObj); err != nil {
+		return
```

Comment:
the will be nil pointer issue when secret is nil and called in `getSecretCACrtAndToken`
you should return err

Reply:
Yes I fixed this in this commit - https://github.com/antrea-io/antrea/pull/3111/commits/fabb15e86e2d70b4780f2878a1e9e9688f217440#diff-3493b4142964cbbdf0ea6689171d62522aef51c3b4f8dbb687070cf0d646fc7d